### PR TITLE
feat(sandbox): declareComponent flag

### PR DESCRIPTION
**Pull request by isaacplmann**
declareComponent flag controls whether the sandbox component is declared automatically - defaults to true

I tested on the angular cli example by setting the notice component to use `declareComponent: false`.  The notice component and the other components all render in the playground without error.

### CODE REVIEW

**Comment by isaacplmann** [comment]
Fixes issue #15

File: src/app/scenario/scenario.component.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export class ScenarioComponent {
         imports ? imports : []
       ],
       declarations: [
-        type,
+        ...(sandbox.declareComponent ? [ type ] : []),
```
Comment (jschwarty) [inline src/app/scenario/scenario.component.ts]: Nice! We should probably change the `declarations ? declarations : []` to use the spread operator too. Can do that in a separate commit.
